### ISSUE TITLE
Create common.erb template for redundant configurations

### ISSUE THREAD

**Issue by chaosmail** [voxpupuli/puppet-nginx#403]
Right now it's a bit annoying, that configurations are avialable in nginx::resource::location but not in nginx::resource::vhost and vice-versa. Creating a common(_header|_footer).erb would facilitate the implementation of common configurations.

**Comment by wyardley** [comment]
I agree that it would be great if a lot of the logic and templating could be made more generic.
Also, the handling of resources that can be multiple types is currently pretty inefficient.
